Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Instance/InstanceDomGenerator.cpp

```diff
@@ -200,38 +202,57 @@ namespace AzToolsFramework
+            for (auto instanceIter = climbedInstances.crbegin(); instanceIter != climbedInstances.crend(); ++instanceIter)
             {
+                // Create a path from the current top instance to the container entity of the given instance.
+                AZStd::string relativePathFromTopInstance =
+                    PrefabInstanceUtils::GetRelativePathBetweenInstances(*topInstance, instance);
+                const AZStd::string pathToContainerEntity =
+                    AZStd::string::format("%s/%s", relativePathFromTopInstance.c_str(), PrefabDomUtils::ContainerEntityName);
+                PrefabDomPath domPathToContainerEntity(pathToContainerEntity.c_str());
+                const PrefabDom& topInstanceTemplateDom =
+                    m_prefabSystemComponentInterface->FindTemplateDom((*topInstance).GetTemplateId());
                 // DOM reference can't be relied upon if the original DOM gets modified after reference creation.
                 // This scope is added to limit their usage and ensure DOM is not modified when it is being used.
-                const PrefabDom& rootTemplateDom = m_prefabSystemComponentInterface->FindTemplateDom((*rootInstancePtr).GetTemplateId());
-                containerEntityDom.CopyFrom(*domPathToContainerEntity.Get(rootTemplateDom), instanceDom.GetAllocator());
+                {
+                    const auto* containerEntityDomPtr = domPathToContainerEntity.Get(topInstanceTemplateDom);
+                    if (containerEntityDomPtr)
+                    {
+                        containerEntityDom.CopyFrom(*containerEntityDomPtr, instanceDom.GetAllocator());
+                        break;
+                    }
+                }
+
+                topInstance = *instanceIter;
             }
 
-            // Sets the container entity DOM in the given instance DOM.
-            const AZStd::string containerEntityName = AZStd::string::format("/%s", PrefabDomUtils::ContainerEntityName);
-            PrefabDomPath containerEntityPath(containerEntityName.c_str());
-            containerEntityPath.Set(instanceDom, containerEntityDom, instanceDom.GetAllocator());
+            AZ_Assert(containerEntityDom.IsObject(), "Prefab - InstanceDomGenerator::UpdateContainerEntityInDomFromHighestAncestor - "
+                "Couldn't find container entity DOM in any ancestor.");
+            if (containerEntityDom.IsObject())
+            {
+                // Sets the container entity DOM in the given instance DOM.
+                const AZStd::string containerEntityName = AZStd::string::format("/%s", PrefabDomUtils::ContainerEntityName);
+                PrefabDomPath containerEntityPath(containerEntityName.c_str());
+                containerEntityPath.Set(instanceDom, containerEntityDom, instanceDom.GetAllocator());
+            }
         }
 
         void InstanceDomGenerator::GenerateContainerEntityDomFromRootInstance(PrefabDom& containerEntityDom, const Instance& instance) const
```

Comment:
I think we probably need to do similar changes to this function.

Reply:
Yes, good call. I tested this scenario and it also crashed. I've updated the PR to cover this.